kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/concurrent/concurrent.kt

```diff
@@ -42,3 +44,20 @@ fun <A> shouldTimeout(timeout: Long, unit: TimeUnit, thunk: () -> A) {
     throw failure("Expected test to timeout for $timeout/$unit")
   }
 }
+
+/**
+ * Verify that an asynchronous call should timeout
+ */
+suspend fun <A> shouldTimeout(timeout: Long, unit: TimeUnit, thunk: suspend () -> A) {
```

Comment:
Can we add a version that uses kotlinx.time.Duration as well. This one should delegate to that one or vice versa.